src/app/reporting/Engine.cpp

```diff
@@ -439,7 +439,15 @@ CHIP_ERROR Engine::SetDirty(ClusterInfo & aClusterInfo)
         // chunk for read interactions.
         if (handler.IsGeneratingReports() || handler.IsAwaitingReportResponse())
         {
-            handler.SetDirty();
+            for (auto clusterInfo = handler.GetAttributeClusterInfolist(); clusterInfo != nullptr;
+                 clusterInfo      = clusterInfo->mpNext)
+            {
+                if (aClusterInfo.IsAttributePathSupersetOf(*clusterInfo) || clusterInfo->IsAttributePathSupersetOf(aClusterInfo))
```

Comment:
What if we 
Subscribe Wildcard Endpoint, Cluster X

And SetDirty with
Endpoint Y, Wildcard Cluster

This may happen when a new dynamic endpoint was added to the server

Reply:
yep, need extend supersetof function to overlapped for clusterInfoA and clusterInfoB, and need to get the overlapped part, and send report with overlapped part, perhaps in

Reply:
https://github.com/project-chip/connectedhomeip/issues/11960